pkg/aws/eni/node.go

```diff
@@ -808,7 +808,11 @@ func (n *Node) GetUsedIPWithPrefixes() int {
 		var pfx string
 		// Check if the IP is from any of the prefixes attached to this ENI
 		for _, ipNet := range prefixNetworks {
-			if ipNet.Contains(net.ParseIP(ip)) {
+			addr, err := netip.ParseAddr(ip)
+			if err != nil {
+				continue
+			}
+			if ipNet.Contains(addr) {
```

Comment:
Maybe while at it, instead of re-parsing the same `ip` for each `ipNet`, how about moving this out of the inner loop?

```go
		addr, err := netip.ParseAddr(ip)
		if err != nil {
			continue
		}
		for _, ipNet := range prefixNetworks {
			if ipNet.Contains(addr) {
				...
			}
		}
```

Reply:
Updated 👍